src/components/Navbar.js

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useRef } from 'react';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import { useKanban } from '../hooks/useKanban';
+
+export function Navbar() {
+  const { user, signInWithGoogle, signOut } = useAuth();
+  const { isDark, toggleTheme } = useTheme();
+  const { boardTitle, updateBoardTitle, addTask, updateColumn, columns } = useKanban();
+  const [isEditingTitle, setIsEditingTitle] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(boardTitle);
+  const [newTask, setNewTask] = useState('');
+  const [newColumnTitle, setNewColumnTitle] = useState('');
+  const [showForms, setShowForms] = useState(false);
+  const titleInputRef = useRef(null);
+
+  const handleTitleClick = () => {
+    if (user) {
+      setIsEditingTitle(true);
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleChange = (e) => {
+const handleTitleBlur = async () => {
+const debouncedSetLoading = debounce((value) => setIsLoading(value), 300);
+debouncedSetLoading(true);
+  setIsEditingTitle(false);
+  if (editedTitle.trim() && editedTitle !== boardTitle) {
+    try {
+      await updateBoardTitle(editedTitle.trim());
+    } catch (error) {
+      setEditedTitle(boardTitle);
+      console.error('Failed to update board title:', error);
+      // Add user feedback mechanism here
+    }
+  } else {
+    setEditedTitle(boardTitle);
+  }
+};
+      setEditedTitle(boardTitle);
+    }
+  };
+
+  const handleTitleKeyDown = (e) => {
+    if (e.key === 'Enter') {
+      handleTitleBlur();
+    }
+  };
+
+  const handleAddTask = async (e) => {
+    e.preventDefault();
+    if (!user || !newTask.trim()) return;
+
+    const task = {
+id: crypto.randomUUID(),
+      content: newTask,
+      createdAt: new Date()
+    };
+
+    const firstColumnId = Object.values(columns)
+      .sort((a, b) => a.order - b.order)[0]?.id;
+
+    if (firstColumnId) {
+      await addTask(firstColumnId, task);
+      setNewTask('');
+    }
+  };
+
+  const handleAddColumn = async (e) => {
+    e.preventDefault();
+    if (!user || !newColumnTitle.trim()) return;
+
+    const columnId = newColumnTitle.toLowerCase().replace(/\s+/g, '-');
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```javascript
const columnId = `${newColumnTitle.toLowerCase().replace(/\s+/g, '-')}-${crypto.randomUUID()}`;
```
The handleAddColumn function uses a simple string manipulation for generating columnId which could lead to duplicate IDs. Consider using a more robust ID generation method like crypto.randomUUID().



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;